app/src/main/java/com/github/se/eventradar/model/EventsOverviewViewModel.kt

```diff
@@ -1,21 +1,76 @@
 package com.github.se.eventradar.model
 
+import android.util.Log
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.github.se.eventradar.model.event.EventList
-import com.google.firebase.firestore.FirebaseFirestore
-import com.google.firebase.firestore.ktx.firestore
-import com.google.firebase.ktx.Firebase
+import com.github.se.eventradar.model.repository.event.IEventRepository
+import com.github.se.eventradar.model.repository.user.IUserRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
 
-class EventsOverviewViewModel(db: FirebaseFirestore = Firebase.firestore) : ViewModel() {
+@HiltViewModel
+class EventsOverviewViewModel
+@Inject
+constructor(
+    private val eventRepository: IEventRepository,
+    private val userRepository: IUserRepository
+) : ViewModel() {
   private val _uiState = MutableStateFlow(EventsOverviewUiState())
-
   val uiState: StateFlow<EventsOverviewUiState> = _uiState
 
-  private val eventRef = db.collection("events")
+  fun getEvents() {
+    viewModelScope.launch {
+      when (val response = eventRepository.getEvents()) {
+        is Resource.Success -> {
+          _uiState.value =
+              _uiState.value.copy(
+                  eventList =
+                      EventList(
+                          response.data, response.data, _uiState.value.eventList.selectedEvent))
+        }
+        is Resource.Failure -> Log.d("EventsOverviewViewModel", "Error getting events")
+      }
+    }
+  }
 
-  fun getEvents() {}
+  fun getUpcomingEvents(uid: String) {
+    viewModelScope.launch {
+      when (val userResponse = userRepository.getUser(uid)) {
+        is Resource.Success -> {
+          val user = userResponse.data ?: return@launch
```

Comment:
I'm not familiar with the syntax, what does it do?

Reply:
If the user is null,  return@launch instructs the function to return immediately from the launch coroutine scope. This prevents any subsequent operations that would rely on a non-null user object.
After further review, there is no need for ?: return@launch since data is guaranteed non-null on success of the getUser(uid) function.